mass_mailing_unique/models/mail_mass_mailing_contact.py

```diff
@@ -16,9 +16,14 @@ def _check_email_list_ids(self):
             others = lists.mapped('contact_ids') - contact
 
             contact_email = contact.email.strip().lower()
-            other_emails = [e.strip().lower() for e in others.mapped('email')]
+            other_emails = {e.strip().lower() for e in others.mapped('email')}
             if contact_email in other_emails:
+                in_list = others.filtered(lambda o: o.email == contact_email) \
+                    .list_ids.mapped('name')
+                lists = ''.join(map(lambda l: '- {}\n'.format(l), in_list))
                 raise ValidationError(_(
-                    "Cannot have the same email (%s) more"
-                    "than once in the same list." % contact_email
+                    '''Email ({0}) already in mailing list(s):
+                    {1}
+                    Please use a different email address or remove ({0}) from the \
+                    mailing list.'''.format(contact_email, lists)
```

Comment:
This is incorrect, as the email and lists should be formatted after the translation is applied, not before.

Reply:
Yes, and translations must use `%s` format.

Reply:
Are you going to fix the commented points?

Reply:
Yes, working on it ATM!

Reply:
finished and looks like all checks passed...